src/jit/MemoryInl.h

```diff
@@ -1011,3 +1011,322 @@ static void emitStore(sljit_compiler* compiler, Instruction* instr)
     // TODO: sljit_emit_mem for unaligned access
     sljit_emit_op1(compiler, opcode, addr.memArg.arg, addr.memArg.argw, addr.loadArg.arg, addr.loadArg.argw);
 }
+
+#if defined(ENABLE_EXTENDED_FEATURES)
+
+#if (defined SLJIT_32BIT_ARCHITECTURE && SLJIT_32BIT_ARCHITECTURE)
+static void atomicRmwAdd64(uint64_t* shared_p, uint64_t* value, uint64_t* result)
+{
+    std::atomic<uint64_t>* shared = reinterpret_cast<std::atomic<uint64_t>*>(shared_p);
```

Comment:
As far as I know, there is no constructor that takes a pointer and creates an atomic pointer; the interpreter also uses `reinterpret_cast`: https://github.com/Samsung/walrus/blob/69fd93a06259a9294aed3b41f23e55a5d152c1e2/src/runtime/Memory.h#L174

Reply:
These atomic* relative functions are defined as `Memory` member functions, which handle the same atomic operations with memory align checks.
Is it more efficient to use these special helper functions in JITC instead of reusing the Memory's member functions?

Reply:
Since alignment check is already done, it should be a bit faster, but since it is only for 32 bit, and probably rarely used, we could use those as well. Btw, not all 32 bit cpus have 64 bit atomic operations (e.g. RISCV), so I have no idea how they do it.

https://github.com/Samsung/walrus/blob/main/src/runtime/Memory.h#L171

Reply:
I realized we have no access to the memory structure in the callback, we get the absolute address of the memory location. Shall we add static functions into the memory to do this?

Reply:
I see
I think that adding more functions to get the address of the Memory would be complex too, so the current helper functions of JITC seems better :)